tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/object/test_bir_negative.bal

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import bir/objs;
+
+public type Foo distinct isolated client object {
+    public isolated function execute() returns anydata|error;
+};
+
+function f1(objs:Bar bar) {
+    objs:Foo _ = bar;
+}
+
+function f2(objs:Foo foo) {
+    objs:Bar _ = foo;
+}
+
+function f3(objs:Xyz xyz) {
+    objs:Qux _ = xyz;
+}
+
+function f4(objs:Qux qux) {
+    objs:Xyz _ = qux;
+}
+
+function f5(Foo foo) {
+    objs:Foo _ = foo;
```

Comment:
Can we check the other way too? IMO, those are more useful, since when both the types are read from the BIR, flags were ignored for both, so doesn't make a difference?

Reply:
Updated.